Lib/zoneinfo/_common.py

```diff
@@ -0,0 +1,156 @@
+import struct
+
+
+def load_tzdata(key):
+    import importlib.resources
+
+    components = key.split("/")
+    package_name = ".".join(["tzdata.zoneinfo"] + components[:-1])
+    resource_name = components[-1]
+
+    try:
+        return importlib.resources.open_binary(package_name, resource_name)
+    except (ImportError, FileNotFoundError, UnicodeEncodeError) as e:
```

Comment:
the except UnicodeEncodeError is not obvious.. may I ask for what special case ?

Reply:
This will occur if you put non-UTF8 characters in the name, like surrogate characters, add they are not valid package names.

Reply:
Maybe add a comment to explain that ;-)

Reply:
Done.